backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/InvoiceDAO.java

```diff
@@ -53,6 +62,33 @@ public I_C_Invoice createInvoice(String trxName)
 		return InterfaceWrapperHelper.create(Env.getCtx(), I_C_Invoice.class, trxName);
 	}
 
+	@Override
+	public Map<OrderId, InvoiceId> getInvoiceIdsForOrderIds(final List<OrderId> orderIds)
+	{
+		Map<OrderId, InvoiceId> orderIdInvoiceIdMap = new HashMap<OrderId, InvoiceId>();
+
+		for (OrderId orderId : orderIds)
+		{
+			orderIdInvoiceIdMap.put(orderId, InvoiceId.ofRepoId(getByOrderId(orderId).getC_Order_ID()));
```

Comment:
i think there is no point loading them one-by-one.
you can use IQueryBuilder.addInArrayFilter()